src/rust/cryptography-x509/src/common.rs

```diff
@@ -45,6 +45,13 @@ pub enum AlgorithmParameters<'a> {
     #[defined_by(oid::ED448_OID)]
     Ed448,
 
+    // These encodings are only used in SPKI AlgorithmIdentifiers.
+    #[defined_by(oid::EC_OID)]
+    EcNamedCurve(asn1::ObjectIdentifier),
+
+    #[defined_by(oid::RSA_OID)]
+    Rsa(Option<asn1::Null>),
```

Comment:
I wasn't sure if these were the best names for these, but I couldn't think of better ones in the moment. Happy to change to whatever.

Reply:
(Renamed `EcNamedCurve` to just `Ec`, due to https://github.com/pyca/cryptography/pull/9800#issuecomment-1785814705.)